src/__tests__/gameHandlerClass.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { GameHandler } from "../gameHandlerClass";
+import { Player } from "../playerClass";
+import { PokemonFactory } from "../pokemonFactory";
+import { PokemonType } from "../pokemonClass";
+import * as data from "../../assets/poke_data.json";
+
+describe("Test Game Handler class", () => {
+  const pokeData = data.pokemons;
+
+  it("GameHandler.isGameFinished should return false if every player has at least one pokemon", () => {
+    // Given
+    const factory_0 = new PokemonFactory(pokeData);
+    const playerOne_0 = new Player("Wojtek", [
+      factory_0.getPokemonByName("bulbasaur"),
+      factory_0.getPokemonByName("charmander"),
+      factory_0.getPokemonByName("pikachu"),
+    ]);
+    const playerTwo_0 = new Player("Ania", [
+      factory_0.getPokemonByName("squirtle"),
+      factory_0.getPokemonByName("weedle"),
+      factory_0.getPokemonByName("pidgey"),
+    ]);
+
+    // When
+    const isGameFinished_0 = new GameHandler(
+      playerOne_0,
+      playerTwo_0
+    ).isGameFinished();
+
+    // Then
+    expect(isGameFinished_0).toBe(false);
+  });
+
+  it("GameHandler.isGameFinished should return true if all pokemons of one player are dead", () => {
+    // Given
+    const factory_1 = new PokemonFactory(pokeData);
+    const playerOne_1 = new Player("Wojtek", [
+      factory_1.getPokemonByName("bulbasaur"),
+      factory_1.getPokemonByName("charmander"),
+      factory_1.getPokemonByName("pikachu"),
+    ]);
+    playerOne_1.pokemons.forEach((pokemon) =>
+      pokemon.subtractHP(pokemon.maxHP)
+    );
+    const playerTwo_1 = new Player("Ania", [
+      factory_1.getPokemonByName("squirtle"),
+      factory_1.getPokemonByName("weedle"),
+      factory_1.getPokemonByName("pidgey"),
+    ]);
+
+    // When
+    const isGameFinished_1 = new GameHandler(
+      playerOne_1,
+      playerTwo_1
+    ).isGameFinished();
+
+    // Then
+    expect(isGameFinished_1).toBe(true);
+  });
+  it("GameHandler.isGameFinished should return false if only some pokemons of a player are dead", () => {
+    // Given
+    const factory_2 = new PokemonFactory(pokeData);
+    const playerOne_2 = new Player("Wojtek", [
+      factory_2.getPokemonByName("bulbasaur"),
+      factory_2.getPokemonByName("charmander"),
+      factory_2.getPokemonByName("pikachu"),
+    ]);
+    const playerTwo_2 = new Player("Ania", [
+      factory_2.getPokemonByName("squirtle"),
+      factory_2.getPokemonByName("weedle"),
+      factory_2.getPokemonByName("pidgey"),
+    ]);
```

Comment:
you can also create helper methods at the end of the file, like:
```
const getPlayerOne = () => {
 return new Player("Wojtek", [
      factory_2.getPokemonByName("bulbasaur"),
      factory_2.getPokemonByName("charmander"),
      factory_2.getPokemonByName("pikachu"),
    ])
}
```
and in test use only:
```
const playerOne = getPlayerOne();
```

Reply:
the reasoning is that: 
"You don't need to know exact details. You just need to know there are some players with some pokemons"

then, in the `when` section you can modify it a bit (like zero all the pokemon healths)